resources/fetching/fetchers/azure/assets_fetcher.go

```diff
@@ -0,0 +1,139 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package fetchers
+
+import (
+	"context"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"golang.org/x/exp/maps"
+
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib/inventory"
+)
+
+type AzureAssetsFetcher struct {
+	log        *logp.Logger
+	resourceCh chan fetching.ResourceInfo
+	provider   inventory.ServiceAPI
+}
+
+type AzureAsset struct {
+	Type    string
+	SubType string
+	Asset   AzureAssetInfo `json:"asset,omitempty"`
+}
+
+// TODO: Fill this struct with the required fields
+type AzureAssetInfo struct {
+	Id             string         `json:"id,omitempty"`
+	Name           string         `json:"name,omitempty"`
+	Location       string         `json:"location,omitempty"`
+	Properties     map[string]any `json:"properties,omitempty"`
+	ResourceGroup  string         `json:"resource_group,omitempty"`
+	SubscriptionId string         `json:"subscription_id,omitempty"`
+	TenantId       string         `json:"tenant_id,omitempty"`
+}
+
+// TODO: Implement other types
+var AzureAssetTypes = map[string]string{
+	"microsoft.compute/virtualmachines": fetching.AzureVMType,
+	"microsoft.storage/storageaccounts": fetching.AzureStorageAccountType,
+}
+
+func NewAzureAssetsFetcher(log *logp.Logger, ch chan fetching.ResourceInfo, provider inventory.ServiceAPI) *AzureAssetsFetcher {
+	return &AzureAssetsFetcher{
+		log:        log,
+		resourceCh: ch,
+		provider:   provider,
+	}
+}
+
+func (f *AzureAssetsFetcher) Fetch(ctx context.Context, cMetadata fetching.CycleMetadata) error {
+	f.log.Info("Starting AzureAssetsFetcher.Fetch")
+	// TODO: Maybe we should use a query per type instead of listing all assets in a single query
```

Comment:
```suggestion
```

Is there any benefit to that?

Reply:
I assume that it might help us If we will need to collect multiple resource types to evaluate a single rule.